src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProject.cs

```diff
@@ -1580,7 +1580,7 @@ private DocumentInfo CreateDocumentInfoFromFileInfo(DynamicFileInfo fileInfo, IE
                 var documentId = DocumentId.CreateNewId(_project.Id, filePath);
 
                 var textLoader = fileInfo.TextLoader;
-                var documentServiceProvider = fileInfo.DocumentServiceProvider;
+                var documentServiceProvider = new DynamicFileDocumentServiceProvider(fileInfo.DocumentServiceProvider);
```

Comment:
Can we file a bug to the ASP.NET team to ultimately implement the document properties service and remove this? I realize you don't have a good answer for this PR but this is something we probably want to undo...

Reply:
Aren't we looking into redesigning this whole thing anyways?